apis/nucleus/src/components/listbox/ListBox.jsx

```diff
@@ -193,6 +199,21 @@ export default function ListBox({
     scrollToIndex(scrollIndex);
   };
 
+  useEffect(() => {
+    const s = keyScroll.state;
+    if (s.up) {
+      scrollToIndex(currentScrollIndex.state.start - s.up);
+    } else if (s.down) {
+      scrollToIndex(currentScrollIndex.state.stop + s.down);
+    } else if (s.scrollPosition) {
+      setScrollPosition(s.scrollPosition);
+    }
+    if (s.scrollPosition === 'overflowEnd') {
+      return; // Do keyScroll.reset() in fetchData() to avoid scrolling to top.
+    }
+    keyScroll.reset();
+  }, [JSON.stringify(keyScroll.state)]);
```

Comment:
Don't know if it's any better… you decide:
```suggestion
  }, [Object.values(keyScroll.state)]);
```

Reply:
Yes that's probably best, it just hurts my dx eyes a bit.